### ISSUE TITLE
FileOpenPicker, FolderPicker and FolderPicker not available in partial trust/AppContainer

### ISSUE THREAD

**Issue by sungaila** [microsoft/microsoft-ui-xaml#9557]
### Describe the bug

In WinUI 3 it is needed to do the following call to initialize a picker dialog:
```csharp
WinRT.Interop.InitializeWithWindow.Initialize(openPicker, hWnd);
```

It works just fine in full trust mode but fails when it is called in partial trust/inside an AppContainer (access denied exception).

This is terrible news because it means that WinUI 3 can either use the WinRT file dialogs OR run in partial trust. But NOT both at the same time.

### Steps to reproduce the bug

1. Create a packaged WinUI 3 app with the default template
2. Change `Package.appxmanifest` such that the app runs in partial mode
3. Optional: add the `broadFileSystemAccess` capability
4. Test the FilePicker code snippet from the WinUI 3 Gallery: [winui3gallery://item/FilePicker](winui3gallery://item/FilePicker)

### Expected behavior

The file dialogs should open just as expected.

### NuGet package version

WinUI 3 - Windows App SDK 1.5.2: 1.5.240404000

### Windows version

Windows 11 (22H2): Build 22621

**Comment by Rybasum** [comment]
This issue is several years old and it looks like Microsoft simply does not want us to put non-UWP apps in AppContainer. Otherwise they would realize that most UI apps use file open/save pickers. AppContainer? yes, sure, but no file open/save. Very strange. Honestly, they should simply document this and not make developers waste their time.

**Comment by sungaila** [comment]
@Rybasum The documentation part is kinda confusing. There is [in-depth documentation on how to put your WPF, WinForms or WinUI 3 app into an AppContainer using MSIX](https://learn.microsoft.com/en-us/windows/msix/msix-container). Yet something fundamental like selecting files isn't possible with Win32 or WinRT APIs.

I have a simple WinUI 3 app here where the user selects a file and more information is put into a DataGrid. You'd think that it would be a perfect candidate for partial trust/AppContainer and just the filesystem capability.

**Comment by mominshaikhdevs** [comment]
@sungaila you want to see https://github.com/microsoft/WindowsAppSDK/issues/219 and also the [win32-app-isolation](https://github.com/microsoft/win32-app-isolation) repo.

**Comment by dyunchik** [comment]
Seriously, who came up with this "brilliant" idea? How could they develop a sandbox without file selection support? Hey, Microsoft, this is truly a shame!

**Comment by Rybasum** [comment]
With WinAppSDK 2.0.1, the pickers still don't work inside AppContainer. The file dialog opens and then crashes with Access Denied in any initial folder. They added support for Pickers when running as Administrator, though. I guess, AppCoontainer should be ready for the next century.

**Comment by charlesroddie** [comment]
@Saharsh979 by closing this issue, are you saying that WinUI is designed to deploy to system administrators only not designed for apps to ordinary end-users?

**Comment by sungaila** [comment]
@Saharsh979 Closing this ticket as "won't fix" is very concerning. The original pull request said "support for AppContainer is not planned yet".

By closing this you say you don't want to support AppContainer at all. WinUI 3 not supporting file pickers for both MediumIL and AppContainer is ... very VERY concerning.

Visual Studio includes a combobox that allows you to choose between MediumIL and AppContainer (see appxmanifest designer). So, of course, one might conclude that WinUI 3 supports both.

**Comment by Rybasum** [comment]
Yes, this sounds incredible, but it's true. Microsoft has revealed that they are not interested in enabling AppContainer for WinUI 3 apps. They want to allow everybody full system access in their apps. Utter contempt for security. Indeed, hard to believe.

**Comment by Saharsh979** [comment]
Hi @charlesroddie @sungaila @Rybasum Sorry for the lack of context around this closure. It was propagated from internal tracking through the mirroring pipeline, which made the GitHub action to close this.  I’m coordinating internally to get clarity on the reasoning behind the closure and will share confirmed context here.
